js/app/application.js

```diff
@@ -245,8 +246,23 @@ const Application = new Knowledge.Class({
         });
     },
 
+    initialize_set_map: function () {
+        Eknc.Engine.get_default().query_promise(Eknc.QueryObject.new_from_props({
+            limit: GLib.MAXUINT32,
+            tags_match_all: ['EknSetObject'],
+        }))
+        .then((results) => {
+            SetMap.init_map_with_models(results.models);
+        })
+        .catch(function (error) {
+            logError(error, 'Failed to load sets from database');
+        });
+    },
+
     _ensure_controller: function () {
         if (this._controller === null) {
+            this.initialize_set_map();
```

Comment:
right, as it is, this is just trying to make the initialization earlier. I am a bit unsure how to address this properly..

a) re. making it sync, it looks all our API in ekncontent is async, so I am not sure how we can make that call sync without modifying the API.

b) re. making this whole block async, still not sure how that would work, since we expecting the controller creation tor be sync (e.g. by ensure_controller).

Any suggestions?